### ISSUE TITLE
resolved: TXT records vanish if /etc/hosts entry is added

### ISSUE THREAD

**Issue by Cornelicorn** [systemd/systemd#33473]
### systemd version the issue has been seen with

256.1

### Used distribution

Arch

### Linux kernel version used

6.9.6-zen1-1-zen

### CPU architectures issue was seen on

x86_64

### Component

systemd-resolved

### Expected behaviour you didn't see

If there is an entry for a host in `/etc/hosts`, any non-address query should be passed to other resolution methods, i.e. if I Add `127.0.2.1 example.com` to `/etc/hosts`, this should not affect a query of the TXT record.

I would still expect:
```
$ resolvectl query -t txt example.com
example.com IN TXT "v=spf1 -all"                            -- link: br0
example.com IN TXT "wgyf8z8cgvm2qmxpnbnldrcltvk4xqfn"       -- link: br0

-- Information acquired via protocol DNS in 476us.
-- Data is authenticated: yes; Data was acquired via local or encrypted transport: no
-- Data from: cache
```

`example.com` is not a placeholder here for once :D

### Unexpected behaviour you saw

```
$ resolvectl query -t txt example.com
example.com: resolve call failed: Name 'example.com' does not have any RR of the requested type
```

### Steps to reproduce the problem

```
echo "127.0.2.1 example.com" | sudo tee -a /etc/hosts
resolvectl query -t txt example.com
```

**Comment by Cornelicorn** [comment]
Checked on Debian12 v252.22: There the query still works when `example.com` is in `/etc/hosts`.

**Comment by poettering** [comment]
Umm, i disagree with this. If you list some domai. in /etc/hosts we really should consider that entry authoritative, and not merge things in a weird way. Sorry, but no.

**Comment by Cornelicorn** [comment]
> Umm, i disagree with this. If you list some domain in /etc/hosts we really should consider that entry authoritative, and not merge things in a weird way. Sorry, but no.

Yes, I also think that `/etc/hosts` should always be the top priority for resolving stuff, but only for stuff that can be resolved with it, which is only a hostname <-> IPv{4,6} address mapping.
In this case it's about querying a TXT record for the host itself. So the conflicting entry is the FQDN entry `x.x.x.x example.com example`. Since a TXT record is by design out-of-scope of `/etc/hosts`, I don't think adding entries to that file should thus change the TXT response.

**Comment by Cornelicorn** [comment]
This would also be the result if you set resolved to `ReadEtcHosts=no` and use `hosts: files dns` in `/etc/nsswitch.conf`.
(`host example.com`  would query resolved still, but `ping example.com` would resolve using `/etc/hosts`, while non-a and non-aaaa queries would get sent to resolved)

**Comment by poettering** [comment]
Sorry, but no. /etc/hosts/ should be authoritative, I am sure. I am strictly against merging RRs from different sources for the same domain. Sorry.

**Comment by gclinch** [comment]
If merging RRs is not intended, this may be a documentation bug for systemd-resolved's 'Synthetic Records' section which appears to describe merging as the intended effect:

> The mappings defined in /etc/hosts are resolved to their configured addresses and back, but they will not affect lookups for non-address types (like MX).

https://github.com/systemd/systemd/blob/main/man/systemd-resolved.service.xml#L134
